app/lib/shared/task_scheduler.dart

```diff
@@ -41,16 +36,37 @@ class TaskScheduler {
   TaskScheduler(this.taskRunner, this.sources);
 
   Future run() async {
-    final Stream<Task> stream =
-        StreamGroup.merge(sources.map((source) => source.startStreaming()));
+    final StreamController newTaskTrigger = new StreamController();
+
+    final List<Queue<Task>> priorityQueues =
+        new List.generate(sources.length, (_) => new Queue());
+
+    // Listen on the streams and put tasks into their own queues.
+    for (int i = 0; i < sources.length; i++) {
+      final TaskSource source = sources[i];
+      final Queue<Task> queue = priorityQueues[i];
+      source.startStreaming().listen((Task task) {
+        queue.add(task);
+        newTaskTrigger.add(null);
+      });
+    }
```

Comment:
Consider encapsulating this functionality into it's own function/class, just like `StreamGroup.merge`, maybe `class PrioritizedStreamGroup<T> {}`.